### PR TITLE
chore: add browser example, JS browser tests, bundling 

### PR DESCRIPTION
**Pull request by SwenSchaeferjohann**

### CODE REVIEW

**Review by SwenSchaeferjohann — COMMENTED** [review]
**Summary of changes**

Notable:
* New `@lightprotocol/test-helpers` lib which bundles test-rpc and related utils 
(test-utils is used as devdependency, and Is temporarily used for merkleproof creation until photon's is fixed)

* Expanded test coverage to long chains of txs (helped uncover bugs)

* Removed hasher.rs from core bundle. (not required for default use)

* Added Prelim browser support (not integrated in CI yet) (nextjs browser app in `examples/`

Misc:
* added (temp) permissive CORS to gnark prover for browser support 
* various fixes (conversion methods,etc..) 
* cleanup (call signatures, etc..)
* updated utils (new `buildtx`, `newaccountwithlamports`-> default random)
* NPM readmes for initial release
* Added remaining RPC methods to TestRPC

File: examples/browser/nextjs/src/app/favicon.ico
Comment (ananas-block) [inline examples/browser/nextjs/src/app/favicon.ico:1]: do we need to commit this?

File: js/stateless.js/src/programs/compressed-pda.ts [hunk 1]
```diff
@@ -407,28 +400,29 @@ export class LightSystemProgram {
...
+    static async decompress({
+        payer,
+        inputCompressedAccounts,
+        toAddress,
+        lamports,
+        recentInputStateRootIndices,
+        recentValidityProof,
+        outputStateTree,
+    }: DecompressParams): Promise<TransactionInstruction> {
+        // const { payer, outputStateTree, toAddress } = params;
```
Comment (ananas-block) [inline js/stateless.js/src/programs/compressed-pda.ts:420]: leftover?

File: js/stateless.js/src/test-helpers/test-rpc/test-rpc.ts [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+    proverEndpoint: string = 'http://localhost:3001',
+    lightWasm?: LightWasm,
+    merkleTreeAddress?: PublicKey,
+    nullifierQueueAddress?: PublicKey,
+    depth?: number,
+    log = false,
+) {
+    console.log('HI');
+    lightWasm = lightWasm || (await WasmFactory.getInstance());
+    console.log('WASM LOADED', lightWasm);
```
Comment (ananas-block) [inline js/stateless.js/src/test-helpers/test-rpc/test-rpc.ts:69]: printsssss

File: js/stateless.js/src/programs/compressed-pda.ts (hunk 1 above)
Comment (SwenSchaeferjohann) [reply js/stateless.js/src/programs/compressed-pda.ts:420]: yep

**Review by SwenSchaeferjohann — COMMENTED** [review]
had another look myself - g2g IMO

File: js/stateless.js/src/rpc.ts [hunk 3]
```diff
@@ -271,26 +326,32 @@ export class Rpc extends Connection implements CompressionApiInterface {
...
 
         const merkleProofs: MerkleContextWithMerkleProof[] = [];
 
-        res.result.value.map((proof: any) => {
+        const rootIndex = await getRootSeq(this);
+
+        for (const proof of res.result.value) {
+            const proofWithoutRoot: BN[] = proof.proof.slice(0, -1);
+
+            // const root = proof.proof[proof.proof.length - 1];
```
Comment (ananas-block) [inline js/stateless.js/src/rpc.ts:341]: leftover

**Review by ananas-block — APPROVED** [review]

**Pull request merged**